zerver/actions/reactions.py

```diff
@@ -181,10 +172,6 @@ def check_add_reaction(
         # realm emoji).
         check_emoji_request(user_profile.realm, emoji_name, emoji_code, reaction_type)
 
-    if user_message is None:
-        # See called function for more context.
-        create_historical_user_messages(user_id=user_profile.id, message_ids=[message.id])
```

Comment:
I don't think we want to remove this logic; if you react to a message in a public stream you're previewing but not subscribed to, we want you to get the reaction, and removing this logic breaks that.

Reply:
I thought it's fine to consider this statement [mentioned in CZO](https://chat.zulip.org/#narrow/stream/3-backend/topic/Tighten.20access.20while.20sending.20events/near/1948785) in this case too:
> We don't currently have a goal of doing live-update systematically for content in channels you're not subscribed to

But, yeah this is a special case. So, dropped that commit which was removing this logic.